File: Practice/20/C++/20.cpp
#include <iostream>
#include <string>
using namespace std;

int main()
{
	int cash(0);
	int number(0);

	cin >> cash;
	cin >> number;

	string* drink = new string[number];
	int* price = new int[number];
	double* volume = new double[number];
	int sumVolume(0);

	int bestChoice(0);

	for (int i = 0;i < number;i++)
	{
		cin >> drink[i] >> price[i] >> volume[i];
	}

	while (true)
	{
		bool canBuy(0);
		for (int i = 0;i < number;i++)
		{
			if (cash >= price[i])
			{
				canBuy = true;
				if (volume[bestChoice] / price[bestChoice] < volume[i] / price[i])
				{
					bestChoice = i;
				}
			}
		}
		if (canBuy == false)
			break;
		cout << drink[bestChoice] << " " << cash / price[bestChoice] << " ";
		sumVolume += (cash / price[bestChoice]) * volume[bestChoice];
		cash %= price[bestChoice];
		volume[bestChoice] = 0;
	}
	if (sumVolume == 0)
		cout << -1;
	else
		printf("\n%d\n%d", sumVolume, cash);

	delete[] drink;
	delete[] price;
	delete[] volume;

}

File: Practice/20/Python/20.py
cash = int(input())
number = int(input())
drink = []
price = []
volume = []
sumVolume = 0
bestChoice = 0
for i in range(0,number):
    idrink,iprice,ivolume = input().split()
    drink.append(idrink)
    price.append(int(iprice))
    volume.append(int(ivolume))
while (True):
    canBuy = False
    for i in range(0,number):
        if cash >= price[i]:
            canBuy = True
            if volume[bestChoice] / price[bestChoice] < volume[i] / price[i]:
                bestChoice = i
    if canBuy == False:
            break;
    print(drink[bestChoice],cash // price[bestChoice])
    sumVolume += cash // price[bestChoice] * volume[bestChoice]
    cash %= price[bestChoice]
    volume[bestChoice] = 0
if sumVolume == 0:
    print("-1")
else:
    print(sumVolume,cash,sep = "\n")
